app/assets/javascripts/models/beta/iteration.js

```diff
@@ -156,52 +196,56 @@ const pastAverageVelocity = (pastIterations) => {
 }
 
 const addToSprintFromBacklog = (sprints, project, story, velocity) => {
-  const sprintIndex = sprints.length && sprints.length - 1;
-  const hasSpace = canTakeStory(sprints[sprintIndex], Story.getPoints(story));
-
-  if (hasSpace) {
-    return addStoryToSprint(project, sprints, sprintIndex, story, velocity);
-  }
-
   const iterationNumber = getNextIterationNumber(sprints);
 
-  sprints[sprintIndex + 1] = createSprint(
+  const newSprint = createSprint(
     iterationNumber,
     getDateForIterationNumber(iterationNumber, project),
-    undefined,
-    velocity
+    false,
+    velocity,
+    [story],
+    Story.getPoints(story)
   );
-
-  return addStoryToSprint(project, sprints, sprintIndex + 1, story, velocity);
+  sprints.push(newSprint);
```

Comment:
I think you could remove `sprints.push(newSprint);` to avoid side effects (mutating `sprints`) and returns a new sprints array with new sprint in it:
```js
return [...sprints, newSprint];
```

Reply:
Oh sure, change directly the array it's not good practice.